corehq/apps/saved_reports/models.py

```diff
@@ -691,84 +695,141 @@ def _get_or_create_uuid(self):
-                if excel_files:
-                    message = _("Unable to generate email report. Excel files are attached.")
-                    send_html_email_async(title, email, message,
-                                          email_from=settings.DEFAULT_FROM_EMAIL,
-                                          file_attachments=excel_files,
-                                          smtp_exception_skip_list=LARGE_FILE_SIZE_ERROR_CODES)
-
-                elif getattr(er, 'smtp_code', None) in LARGE_FILE_SIZE_ERROR_CODES or type(er) == ESError:
-                    # If the email doesn't work because it is too large to fit in the HTML body,
-                    # send it as an excel attachment, by creating a mock request with the right data.
-
-                    for report_config in self.configs:
-                        mock_request = HttpRequest()
-                        mock_request.couch_user = self.owner
-                        mock_request.user = self.owner.get_django_user()
-                        mock_request.domain = self.domain
-                        mock_request.couch_user.current_domain = self.domain
-                        mock_request.couch_user.language = self.language
-                        mock_request.method = 'GET'
-                        mock_request.bypass_two_factor = True
-
-                        mock_query_string_parts = [report_config.query_string, 'filterSet=true']
-                        mock_request.GET = QueryDict('&'.join(mock_query_string_parts))
-                        request_data = vars(mock_request)
-                        request_data['couch_user'] = mock_request.couch_user.userID
-                        if report_config.report_slug != ApplicationStatusReport.slug:
-                            # ApplicationStatusReport doesn't have date filter
-                            date_range = report_config.get_date_range()
-                            start_date = datetime.strptime(date_range['startdate'], '%Y-%m-%d')
-                            end_date = datetime.strptime(date_range['enddate'], '%Y-%m-%d')
-                            datespan = DateSpan(start_date, end_date)
-                            request_data['datespan'] = datespan
-
-                        full_request = {'request': request_data,
-                                        'domain': request_data['domain'],
-                                        'context': {},
-                                        'request_params': json_request(request_data['GET'])}
-
-                        export_all_rows_task(report_config.report, full_request, emails, title)
+                if isinstance(er, SMTPSenderRefused) and (er.smtp_code in LARGE_FILE_SIZE_ERROR_CODES):
+                    # This should only occur on the first error we try to send,
+                    # as all the emails should be the same size
+                    assert total_sent == 0, f'Email sizes inconsistent, failed on email #{total_sent} in {emails}'
```

Comment:
Is it possible that different email domains would have different size limits, so the first email may send successfully and a later one may fail on account of size?

Reply:
Interesting thought. I'm not sure how email negotiation works, but that's probably true? I'll look into it.

Reply:
After investigation, this is a valid concern, but also something that was part of our existing codebase -- I just made an explicit assertion about it. I've left a `TODO` item to revisit this in the future, and just removed the assertion. I think the correct way to handle this is likely to attempt to send each email individually, recording which failed, and only re-sending those.